scribejava-apis/src/main/java/com/github/scribejava/apis/microsoftazureactivedirectory/BaseMicrosoftAzureActiveDirectoryApi.java

```diff
@@ -30,6 +30,11 @@ protected String getAuthorizationBaseUrl() {
         return MSFT_LOGIN_URL + tenant + OAUTH_2 + getEndpointVersionPath() + "/authorize";
     }
 
+    @Override
+    public String getDeviceAuthorizationUrl() {
+        return MSFT_LOGIN_URL + tenant + OAUTH_2 + getEndpointVersionPath() + "/devicecode";
+    }
```

Comment:
I have added Google implementation to the master, because I can test it. I have app and user.
Provide me with MS test app credentials, test user credentials, please, so I can test them before merging to the master.

Example as https://github.com/scribejava/scribejava/blob/master/scribejava-apis/src/test/java/com/github/scribejava/apis/examples/Google20DeviceAuthorizationGrantExample.java is needed as well, but it's easy to create (almost copy-paste).